src/com/seafile/seadroid2/ui/activity/SearchActivity.java

```diff
@@ -0,0 +1,423 @@
+
+        }
+
+        @Override
+        public void onTextChanged(CharSequence s, int start, int before,
+                                  int count) {
+            if (mTextField.getText().toString().length() > 0) {
+                mTextClearBtn.setVisibility(View.VISIBLE);
+                mSearchBtn.setVisibility(View.VISIBLE);
+            } else {
+                mTextClearBtn.setVisibility(View.GONE);
+                mSearchBtn.setVisibility(View.GONE);
+            }
+        }
+
+        @Override
+        public void afterTextChanged(Editable s) {
+
+        }
+    }
+
+    class EditorActionListener implements TextView.OnEditorActionListener {
+
+        @Override
+        public boolean onEditorAction(TextView v, int actionId,
+                                      KeyEvent event) {
+            if (actionId == EditorInfo.IME_ACTION_SEARCH
+                    || (event != null && event.getKeyCode() == KeyEvent.KEYCODE_ENTER)) {
+                // pass 0 to disable page loading
+                handleSearch(0);
+                return true;
+            }
+            return false;
+        }
+    }
+
+    public void onSearchedFileSelected(SearchedFile searchedFile) {
+        final String repoID = searchedFile.getRepoID();
+        SeafRepo seafRepo = dataManager.getCachedRepoByID(repoID);
+        final String repoName = seafRepo.getName();
+        final String filePath = searchedFile.getPath();
+
+        if (searchedFile.isDir()) {
+            if (seafRepo == null) {
+                showToast("Couldn't find this library. It may be deleted");
+                return;
+            }
+            showRepo(repoID, repoName, filePath, null);
+            //showToast(R.string.search_open_dir_not_support);
+            return;
+        }
+
+        File localFile = dataManager.getLocalCachedFile(repoName, repoID, filePath, null);
+        if (localFile != null) {
+            showFile(localFile);
+            return;
+        }
+
+        startFileActivity(repoName, repoID, filePath);
+    }
+
+    private void startFileActivity(String repoName, String repoID, String filePath) {
+        int taskID = txService.addDownloadTask(account, repoName, repoID, filePath);
+        Intent intent = new Intent(this, FileActivity.class);
+        intent.putExtra("repoName", repoName);
+        intent.putExtra("repoID", repoID);
+        intent.putExtra("filePath", filePath);
+        intent.putExtra("account", account);
+        intent.putExtra("taskID", taskID);
+        startActivity(intent);
+    }
+
+    private void showRepo(String repoID, String repoName, String path, String dirID) {
+        Intent intent = new Intent(this, BrowserActivity.class);
+        intent.putExtra("repoID", repoID);
+        intent.putExtra("repoName", repoName);
+        intent.putExtra("path", path);
+        intent.putExtra("dirID", dirID);
+        startActivity(intent);
+    }
+
+    public void showFile(File file) {
```

Comment:
I see this method and the `startMarkdownActivity` method below are copied from `FileActivity` class, can we refactor them out to a helper function?